pythonforandroid/graph.py

```diff
@@ -7,49 +7,79 @@
 from pythonforandroid.util import BuildInterruptingException
 
 
-class RecipeOrder(dict):
+def fix_deplist(deps):
+    """ Turn a dependency list into lowercase, and make sure all entries
+        that are just a string become a tuple of strings
+    """
+    deps = [
+        ((dep.lower(),) if not isinstance(dep, (list, tuple))
+         else tuple([dep_entry.lower() for dep_entry in dep]))
+        for dep in deps
+    ]
+    return deps
+
 
+class RecipeOrder(dict):
     def __init__(self, ctx):
         self.ctx = ctx
 
-    def conflicts(self, name):
+    def conflicts(self):
         for name in self.keys():
             try:
                 recipe = Recipe.get_recipe(name, self.ctx)
-                conflicts = recipe.conflicts
-            except IOError:
+                conflicts = [dep.lower() for dep in recipe.conflicts]
+            except ValueError:
                 conflicts = []
 
             if any([c in self for c in conflicts]):
                 return True
         return False
 
 
-def recursively_collect_orders(name, ctx, all_inputs, orders=[]):
+def get_dependency_tuple_list_for_recipe(recipe, blacklist=[]):
+    """ Get the dependencies of a recipe with filtered out blacklist, and
+        turned into tuples with fix_deplist()
+    """
+    if recipe.depends is None:
+        dependencies = []
+    else:
+        # Turn all dependencies into tuples so that product will work
+        dependencies = fix_deplist(recipe.depends)
+
+        # Filter out blacklisted items and turn lowercase:
+        dependencies = [
+            deptuple for deptuple in dependencies
+            if len(set(deptuple).intersection(set(blacklist))) == 0
```

Comment:
`if not len(...)` or even `if not ...` only.

Reply:
changed it, thanks :+1: